File: deskover-ecommerce-website/src/main/java/com/deskover/entity/Order.java
package com.deskover.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Timestamp;
import java.util.LinkedHashSet;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "orders")
public class Order implements Serializable {
	
    private static final long serialVersionUID = 3887345717822508619L;
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "order_code", nullable = false, length = 11)
    private String orderCode;
    
    @Column(name = "order_qr_code")
    private String qrCode;
    
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "payment_id")
    private PaymentMethods payment;
    
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "shipping_id")
    private ShippingMethods shipping;
    
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "status_id")
    private OrderStatus orderStatus;
    
    @Column(name="note")
    private String note;
    
    @Column(name="shipping_note")
    private String shipping_note;

    @Column(name = "full_name", nullable = false, length = 128)
    private String fullName;
    
    @Column(name = "email", length = 50)
    private String email;

    @Column(name = "created_at", nullable = false)
    private Timestamp createdAt;

    @Column(name = "modified_by", length = 50)
    private String modifiedBy;
    
    @OneToMany(mappedBy = "order")
    private Set<OrderDetail> orderDetails = new LinkedHashSet<>();

    @OneToMany(mappedBy = "order")
    private Set<OrderItem> orderItems = new LinkedHashSet<>();

}

File: src/app/entites/order.ts
import {Customer} from "@/entites/customer";
import {Payment} from "@/entites/payment";
import {Shipping} from "@/entites/shipping";
import {Product} from "@/entites/product";

export interface Order {
  id: number
  orderCode: string
  qrCode: string
  user: Customer
  payment: Payment
  shipping: Shipping
  orderStatus: OrderStatus
  statusPayment: StatusPayment
  note: string
  shipping_note: string
  fullName: string
  email: string
  createdAt: string
  modifiedBy: string
  unitPrice: number
  orderQuantity: number
  label: string
  fee: number
  estimated_pick_time: Date
  estimated_deliver_time: Date
  orderDetail: OrderDetail
  products: OrderItem[]
}

export interface StatusPayment {
  id: number
  code: string
  status: string
}

export interface OrderDetail {
  id: number
  address: string
  province: string
  district: string
  ward: string
  tel: string
}

export interface OrderItem {
  id: number;
  quantity: number;
  price: number;
  order: Order;
  product: Product;
}

export interface OrderStatus {
  id: number
  code: string
  status: string
}

